tool/gocross/gocross-wrapper.sh

```diff
@@ -71,15 +60,38 @@ if [[ ! -d "$toolchain" ]]; then
             # Go uses the name "amd64".
             HOST_ARCH="amd64"
         fi
-
-        rm -rf "$toolchain" "$toolchain.extracted"
         curl -f -L -o "$toolchain.tar.gz" "https://github.com/tailscale/go/releases/download/build-${REV}/${HOST_OS}-${HOST_ARCH}.tar.gz"
         mkdir -p "$toolchain"
         (cd "$toolchain" && tar --strip-components=1 -xf "$toolchain.tar.gz")
         echo "$REV" >"$toolchain.extracted"
         rm -f "$toolchain.tar.gz"
-        ;;
-    esac
+
+        # Do some cleanup of old toolchains while we're here.
+        for hash in $(find "$HOME/.cache/tsgo" -type f -maxdepth 1 -name '*.extracted' -mtime 90 -exec basename {} \; | sed 's/.extracted$//'); do
+            echo "# Cleaning up old Go toolchain $hash" >&2
+            rm -rf "$HOME/.cache/tsgo/$hash"
+            rm -rf "$HOME/.cache/tsgo/$hash.extracted"
+        done
+    fi
+    ;;
+esac
+
+if [[ -d "$toolchain" ]]; then
+    # A toolchain exists, but is it recent enough to compile gocross? If not,
+    # wipe it out so that the next if block fetches a usable one.
+    want_go_minor=$(grep -E '^go ' "go.mod" | cut -f2 -d'.')
+    have_go_minor=""
+    if [[ -f "$toolchain/VERSION" ]]; then
+        have_go_minor=$(head -1 "$toolchain/VERSION" | cut -f2 -d'.')
```

Comment:
```suggestion
        have_go_minor="$(head -1 "$toolchain/VERSION" | cut -f2 -d'.')"
```

Reply:
I think this was just movement, but can do.

I hope there's nothing shell-expansion-globby in the VERSION file :)

Reply:
Oh, I'd hope not too—this is mostly just reflexive, like seeing a `name_with_underscores` in a Go program. It's fine, it just triggers my alerts. 😆